CS3219-AY2425S1/cs3219-ay2425s1-project-g11
# Admin Frontend

This PR aims to implement middleware and an admin frontend

### Things added
- Auth store to store the user information and token client side
- Admin middleware to check if the user is an admin
- Changed CORS for question service to allow request from anywhere
- Add serverside axios for serverside requests
- Add Admin CUD page with create, edit and delete operations

### Breaking changes
- Added API versioning to user service, with `/api/v1` prefix before any route

https://github.com/user-attachments/assets/1db035b3-7d7c-47eb-9c13-abfeab248e2b

### How to test
- Do `docker compose up --build`
- Check if middleware by going to `/admin` when not logged in
- Log in with an admin account
- Check if can enter `/admin` route
- Test the create question functionality. 
  - You should not be able to leave any fields blank
- Test the edit question functionality
  - You should not be able to give the question a title that has already be chosen
- Test the delete question functionality.

Closes #34
